layers/modbustcp.go

```diff
@@ -0,0 +1,138 @@
+// Developed by Vitor Graveto
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+//
+//******************************************************************************
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+	"github.com/google/gopacket"
+)
+
+//******************************************************************************
+//
+// ModbusTCP Decoding Layer
+// ------------------------------------------
+// This file provides a GoPacket decoding layer for ModbusTCP.
+//
+//******************************************************************************
+
+const mbapRecordSizeInBytes int = 7
+const modbusPDUMinimumRecordSizeInBytes int = 2
+const modbusPDUMaximumRecordSizeInBytes int = 253
+
+//******************************************************************************
+
+// ModbusTCP Type
+// --------
+// Type ModbusTCP implements the DecodingLayer interface. Each ModbusTCP object
+// represents in a structured form the MODBUS Application Protocol header (MBAP) record present as the TCP
+// payload in an ModbusTCP TCP packet.
+//
+type ModbusTCP struct {
+	BaseLayer // Stores the packet bytes and payload (Modbus PDU) bytes .
+
+	TransactionIdentifier uint16 // Identification of a MODBUS Request/Response transaction
+	ProtocolIdentifier    uint16 // 0 = MODBUS protocol
```

Comment:
should this be an enum type?

Reply:
It could be an enum, but the documentation has no other possible values for it, see chapter 3.3.1 for MBAP Header description at attached file.

<img width="858" alt="mbap" src="https://user-images.githubusercontent.com/11277540/47738819-e0725200-dc6b-11e8-8ce8-5f4757550465.png">

Reply:
On this field some undocumented version may exist and break the decoder if we impose an enum. So if you agree I would leave it this way.

Reply:
Sorry, by "enum" I meant:


```go
type ModbusProtocol uint16
const (
  ModbusProtocolModbus ModbusProtocol = 0
)
```

That way, if we eventually find other protocols, we can add them, but you can always accept any uint16 value.

Reply:
Ok, good idea I will change it.